modules/tinymce/src/core/test/ts/browser/DisabledModeTest.ts

```diff
@@ -0,0 +1,523 @@
+    toolbar: 'bold',
+    statusbar: false
+  };
+
+  context('Disabled mode', () => {
+    const hook = TinyHooks.bddSetup<Editor>({ ...settings, plugins: 'table' }, [ TablePlugin ]);
+
+    const assertNestedContentEditableTrueDisabled = (editor: Editor, state: boolean, offscreen: boolean) => {
+      TinyAssertions.assertContentStructure(editor,
+        ApproxStructure.build((s, str, _arr) => {
+          const attrs = state ? {
+            'contenteditable': str.is('false'),
+            'data-mce-contenteditable': str.is('true')
+          } : {
+            'contenteditable': str.is('true'),
+            'data-mce-contenteditable': str.none()
+          };
+
+          return s.element('body', {
+            children: [
+              s.element('div', {
+                attrs: {
+                  contenteditable: str.is('false')
+                },
+                children: [
+                  s.text(str.is('a')),
+                  s.element('span', {
+                    attrs,
+                  }),
+                  s.text(str.is('c'))
+                ]
+              }),
+              ...offscreen ? [ s.element('div', {}) ] : [] // Offscreen cef clone
+            ]
+          });
+        })
+      );
+    };
+
+    const assertFakeSelection = (editor: Editor, expectedState: boolean) => {
+      assert.equal(editor.selection.getNode().hasAttribute('data-mce-selected'), expectedState, 'Selected element should have expected state');
+    };
+
+    const assertResizeBars = (editor: Editor, expectedState: boolean) => {
+      SelectorFind.descendant(Traverse.documentElement(TinyDom.document(editor)), '.ephox-snooker-resizer-bar').fold(
+        () => {
+          assert.isFalse(expectedState, 'Was expecting to find resize bars');
+        },
+        (bar) => {
+          const actualDisplay = Css.get(bar, 'display');
+          const expectedDisplay = expectedState ? 'block' : 'none';
+          assert.equal(actualDisplay, expectedDisplay, 'Should be expected display state on resize bar');
+        }
+      );
+    };
+
+    const mouseOverTable = (editor: Editor) => {
+      const table = UiFinder.findIn(TinyDom.body(editor), 'table').getOrDie();
+      Mouse.mouseOver(table);
+    };
+
+    const assertToolbarDisabled = (expectedState: boolean) => {
+      const elm = UiFinder.findIn(SugarBody.body(), 'button[data-mce-name="bold"]').getOrDie();
+      assert.equal(Class.has(elm, 'tox-tbtn--disabled'), expectedState, 'Button should have expected disabled state');
+    };
+
+    const assertHrefOpt = (editor: Editor, selector: string, expectedHref: Optional<string>) => {
+      const elm = SugarElement.fromDom(editor.dom.select(selector)[0]);
+      const hrefOpt = Disabled.getAnchorHrefOpt(editor, elm);
+      Assert.eq('href options match', expectedHref, hrefOpt, tOptional());
+    };
+
+    it('TBA: Switching to disabled mode while having cef selection should remove fake selection', () => {
```

Comment:
You don't need TBA here use the real TINY-<N> ticket number or omit it but I assume this is copied from some other test when we still used that odd pattern.